### PR TITLE
funds-manager: execution-client: Add quote execution helpers

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR sets up the execution flow for the `ExecutionClient` which involves:
1. Approving an ERC20 allowance for the 0x exchange proxy
2. Sending the transaction detailed by the quote received

### Testing
- Tested buying $GMX with $USDC on arbitrum using my personal wallet

### CODE REVIEW

File: funds-manager/funds-manager-server/src/execution_client/mod.rs [hunk 1]
```diff
@@ -61,4 +80,12 @@ impl ExecutionClient {
             .await
             .map_err(ExecutionClientError::http)
     }
+
+    /// Get an instance of a signer middleware with the http provider attached
+    fn get_signer(
+        &self,
+        wallet: LocalWallet,
+    ) -> SignerMiddleware<Arc<Provider<Http>>, LocalWallet> {
+        SignerMiddleware::new(self.rpc_provider.clone(), wallet)
```
Comment (akirillo) [inline funds-manager/funds-manager-server/src/execution_client/mod.rs:89]: we could attach the chain ID here for convenience

File: funds-manager/funds-manager-server/src/execution_client/swap.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        self.approve_erc20_allowance(
+            quote.sell_token_address,
+            quote.to,
+            quote.sell_amount,
+            &wallet,
+        )
+        .await?;
+
+        // Execute the swap
+        let receipt = self.execute_swap_tx(quote, wallet).await?;
```
Comment (akirillo) [inline funds-manager/funds-manager-server/src/execution_client/swap.rs:33]: could we revoke the approved allowance if the swap execution fails? would prefer to leave no unused allowance on any external contracts

File: funds-manager/funds-manager-server/src/execution_client/swap.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    async fn execute_swap_tx(
+        &self,
+        quote: ExecutionQuote,
+        wallet: LocalWallet,
+    ) -> Result<TransactionReceipt, ExecutionClientError> {
+        let client = self.get_signer(wallet.clone());
+        let tx = Eip1559TransactionRequest::new()
+            .to(quote.to)
+            .from(quote.from)
+            .value(quote.value)
```
Comment (akirillo) [inline funds-manager/funds-manager-server/src/execution_client/swap.rs:67]: we expect this to always be 0, right? might as well exclude it here (and maybe from the `ExecutionQuote` struct), just to bar against any bugs in the 0x api

**Review by akirillo — APPROVED** [review]

File: funds-manager/funds-manager-server/src/execution_client/swap.rs (hunk 2 above)
Comment (joeykraut) [reply funds-manager/funds-manager-server/src/execution_client/swap.rs:33]: you'll see in a follow up that is not possible as we separate out allowances from execution.

File: funds-manager/funds-manager-server/src/execution_client/swap.rs (hunk 3 above)
Comment (joeykraut) [reply funds-manager/funds-manager-server/src/execution_client/swap.rs:67]: their API does not guarantee this to be zero, i'll go with what they've claimed in their docs

File: funds-manager/funds-manager-server/src/execution_client/mod.rs (hunk 1 above)
Comment (joeykraut) [reply funds-manager/funds-manager-server/src/execution_client/mod.rs:89]: this is done in a follow up on the wallet directly

**Pull request merged**